help.go

```diff
@@ -246,11 +248,13 @@ func ShowCommandHelp(ctx *Context, command string) error {
 	}
 	for _, c := range commands {
 		if c.HasName(command) {
-			if !ctx.App.HideHelpCommand && !c.HasName(helpName) && len(c.Subcommands) != 0 && c.Command(helpName) == nil {
-				c.Subcommands = append(c.Subcommands, helpCommandDontUse)
-			}
-			if !ctx.App.HideHelp && HelpFlag != nil {
-				c.appendFlag(HelpFlag)
+			if !c.HideHelp {
+				if !c.HideHelpCommand && len(c.Subcommands) != 0 && c.Command(helpName) == nil {
+					c.Subcommands = append(c.Subcommands, helpCommandDontUse)
+				}
```

Comment:
@remiposo can you add a test case for app -h help and app cmd -h help etc ?

Reply:
Sure